lmms_eval/api/task.py

```diff
@@ -1114,6 +1109,7 @@ def fewshot_context(
         apply_chat_template: bool = False,
         fewshot_as_multiturn: bool = False,
         chat_template: Optional[Callable] = None,
+        is_multimodal: bool = False,
```

Comment:
Base on my understanding on the changes on this `few_shot_context` calling, this `is_multimodal` currently is always False?

https://github.com/EvolvingLMMs-Lab/lmms-eval/blob/c5abe5723c06499696e1d7cedfba78da7a852625/lmms_eval/api/task.py#L441-L449

Reply:
Yes, it's not called in current lmms-eval standard pipeline, but in another project it's called via importing from lmms_eval. 

The actual code is following.

```python
visual = self.task_obj.doc_to_visual(self.hf_dataset[index])
if self.shot_number > 0 and (self.task_group_name in ["mmlu_flan_n_shot_generative"] or self.task_name in ["mmlu_flan_n_shot_local_all", "textvqa_val_fewshot", "textcaps_val_fewshot"]):
    if self.task_name in ["mmlu_flan_n_shot_local_all"]:
        is_multimodal = False
    else:
        is_multimodal = True
    context, multimodal_ctx = self.task_obj.fewshot_context(self.hf_dataset[index], num_fewshot=self.shot_number, system_instruction="", is_multimodal=is_multimodal)
    context = context.strip("\n\n")
    if isinstance(multimodal_ctx, list) and isinstance(visual, list):
        visual = multimodal_ctx + visual
    else:
        raise ValueError(f"multimodal_ctx: {multimodal_ctx} is not a list, please check the task_obj.fewshot_context")
else:
    context = self.task_obj.doc_to_text(self.hf_dataset[index])
```

Reply:
So I set it to WIP and wish if we can altogether refine this PR lol. 

I think Fanyi @pufanyi did similar things many months ago.